packages/@uppy/utils/src/RateLimitedQueue.js

```diff
@@ -149,6 +159,53 @@ class RateLimitedQueue {
       return outerPromise
     }
   }
+
+  resume () {
+    this.#paused = false
+    clearTimeout(this.#pauseTimer)
+  }
+
+  #resume = () => this.resume()
+
+  /**
+   * Freezes the queue for a while or indefinitely.
+   *
+   * @param {number | null } [duration] Duration for the pause to happen, in milliseconds.
+   *                                    If omitted, the queue won't resume automatically.
+   */
+  pause (duration = null) {
+    this.#paused = true
+    clearTimeout(this.#pauseTimer)
+    if (duration != null) {
+      this.#pauseTimer = setTimeout(this.#resume, duration)
+    }
+  }
+
+  rateLimit (duration) {
+    clearTimeout(this.#rateLimitingTimer)
+    this.pause(duration)
+    if (this.limit > 1 && Number.isFinite(this.limit)) {
+      this.#upperLimit = this.limit - 1
+      this.limit = this.#downLimit
+      this.#rateLimitingTimer = setTimeout(this.#increaseLimit, duration)
+    }
+  }
```

Comment:
I think a comment would help here on what the strategy is with increase limit. Currently takes a dig into the code to understand what rate limit does, and why it isn't simply a timer increase.